### ISSUE TITLE
Support  Sublime Music

### ISSUE THREAD

**Issue by dadall** [owncloud/music#915]
Hey there,

Could it be possible to make Music compatible with [Sublime Music ](https://gitlab.com/sublime-music/sublime-music)?

Trying to add Nextcloud as a subsonic server leads to this error : 

```
2021-11-09 10:59:44,695:WARNING:root:adapter:Salted auth not supported, falling back to regular password auth
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.8/site-packages/sublime_music/app.py", line 167, in do_activate
    self.show_configure_servers_dialog()
  File "/home/user/.local/lib/python3.8/site-packages/sublime_music/app.py", line 1019, in show_configure_servers_dialog
    self.app_config.save()
  File "/home/user/.local/lib/python3.8/site-packages/sublime_music/config.py", line 234, in save
    json = self.to_json(indent=2, sort_keys=True)
  File "/home/usef/.local/lib/python3.8/site-packages/dataclasses_json/api.py", line 49, in to_json
    return json.dumps(self.to_dict(encode_json=False),
  File "/home/user/.local/lib/python3.8/site-packages/dataclasses_json/api.py", line 85, in to_dict
    return _asdict(self, encode_json=encode_json)
  File "/home/user/.local/lib/python3.8/site-packages/dataclasses_json/core.py", line 346, in _asdict
    return _encode_overrides(dict(result), _user_overrides_or_exts(obj),
  File "/home/user/.local/lib/python3.8/site-packages/dataclasses_json/core.py", line 110, in _encode_overrides
    v = encoder(v) if encoder is not None else v
  File "/home/user/.local/lib/python3.8/site-packages/sublime_music/config.py", line 71, in encode_providers
    return {
  File "/home/user/.local/lib/python3.8/site-packages/sublime_music/config.py", line 74, in <dictcomp>
    "ground_truth_adapter_type": config["ground_truth_adapter_type"].__name__,
TypeError: 'ProviderConfiguration' object is not subscriptable
```

Best regards and thanks of lot for the Music App !

**Comment by paulijar** [comment]
Have you tried reporting this to Sublime Music? While it is totally possible that the root cause is Music app doing something wrong, the Sublime client seems to be lacking in error handling, at least. That doesn't look like graceful error handling, and it doesn't give any hint about what is actually wrong. 

But what I can tell from that print out is that Sublime notices that the server doesn't support token+salt authentication (which is correct) and then tries to use the password authentication as it should. But then it apparently shows some configuration dialog but saving the settings of that dialog fails. This sounds more like a client bug than a server bug, but who knows.

**Comment by dadall** [comment]
Fixed with the last version (v0.11.16) of Sublime.
